prov/verbs/src/verbs_info.c

```diff
@@ -1038,18 +1037,14 @@ static int fi_ibv_fill_addr(struct rdma_addrinfo *rai, struct fi_info **info,
 	return 0;
 }
 
-int fi_ibv_init_info(void)
+int fi_ibv_init_info(struct fi_info **raw_info)
 {
 	struct ibv_context **ctx_list;
 	struct fi_info *fi = NULL, *tail = NULL;
 	int ret = 0, i, num_devices, fork_unsafe = 0;
 
-	if (verbs_info)
-		return 0;
-
 	pthread_mutex_lock(&verbs_info_lock);
```

Comment:
This lock wouldn't be needed anymore since we are not working with a global variable.